pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/RecordReaderFileConfig.java

```diff
@@ -34,8 +34,10 @@ public class RecordReaderFileConfig {
   public final File _dataFile;
   public final Set<String> _fieldsToRead;
   public final RecordReaderConfig _recordReaderConfig;
+  private final boolean _canModifyRecordReader;
```

Comment:
Can you explain what this means ?  May be we can name this _isDelegateReader to indicate that the client has passed in its own recordReader ?

Reply:
changed. I have mentioned the closing logic where we close the recordreader in the comments.